### PR TITLE
test: [M3-6169] - Account cancellation integration tests

### PR DESCRIPTION
**Pull request by jdamore-linode**
## Description 📝
This adds Cypress integration tests for various UI flows related to account cancellation via Cloud Manager's Account Settings page. Broadly speaking, these tests confirm that users can cancel their account, are correctly prompted to complete an exit survey, and that Cloud responds gracefully to API errors that occur during the cancellation process.

## Changes  🔄
- Adds account cancellation test for unrestricted users
- Adds account cancellation test for restricted users
  - ℹ️ Currently Cloud doesn't handle this any differently from any other API error that could occur when cancelling an account, but if we were to enhance this UX we'll now have a test case to cover it

## How to test 🧪
We can rely on CI to make sure that the tests pass and that changes made to mock utils and UI helpers do not cause any unintended issues.

To run the test locally, you can run`yarn && yarn build && yarn start:manager:ci` and then run:

```bash
yarn cy:run -s "cypress/e2e/core/account/account-cancellation.spec.ts"
```

Alternatively, to run the tests interactively you can use `yarn cy:debug` and search for `account-cancellation.spec.ts`.

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/cypress/support/intercepts/account.ts [hunk 1]
```diff
@@ -322,3 +328,39 @@ export const mockGetPayments = (
...
+ *
+ * @param errorMessage - Error message to include in mock error response.
+ * @param status - HTTP status for mock error response.
+ *
+ * @returns Cypress chainable.
+ */
+export const mockCancelAccountError = (
+  errorMessage: string,
+  status: number = 400
+) => {
```
Comment (dwiley-akamai) [inline packages/manager/cypress/support/intercepts/account.ts:360]: Should a `Cypress.Chainable` return type be added for this one as well?

**Review by dwiley-akamai — APPROVED** [review]
Code review ✅
`account-cancellation.spec.ts` passes locally ✅

**Review by cpathipa — APPROVED** [review]
LGTM!
image

File: packages/manager/cypress/support/intercepts/account.ts (hunk 1 above)
Comment (jdamore-linode) [reply packages/manager/cypress/support/intercepts/account.ts]: @dwiley-akamai Yep, thanks for catching that!

**Pull request merged**